src/assembler/parser/index.ts

```diff
@@ -0,0 +1,92 @@
+import { ICodeFile, AreaType, IArea, IInstruction } from 'assembler/ast'
+
+import { ParseError } from './error'
+import { parseText, ITextParseRule, ITextMatch } from './text'
+
+/**
+ * Parses a given code string and return a parsed code file (AST representation).
+ *
+ * @param code the assembler code (text representation)
+ * @returns the parsed code file (AST representation)
+ */
+export function parse(code: string): ICodeFile {
+  const ast: ICodeFile = {
+    symbols: {},
+    areas: []
+  }
+  let area: IArea | null = null
+  let label: string | null = null
+  const rules: ITextParseRule[] = [
```

Comment:
Really cool structure of first defining the rules and afterwards let the parser do the job with the given array of rules. ^^

Reply:
Thanks :blush:

I had to come up with an alternative to the `lezer` parser system :sweat_smile:. It's a quite simple regex based text parser.